<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>EMIM: Estimation of Maternal, Imprinting and interaction effects using Multinomial modelling</title>
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1" />
<link rel="stylesheet" type="text/css" href="emimstyles.css" />
<link rel="shortcut icon" href="favicon.ico" />
</head>
<body>
<!-- Begin Wrapper -->
<div id="wrapper">
<table width="100%" border="0" cellpadding="0" cellspacing="0"><tr><td>
<!-- Begin Header -->
<div id="header">
<table><tr>
<td>
<img width="200" src="emim5.png" alt="EMIM: Estimation of Maternal, Imprinting and interaction effects using Multinomial modelling"/>
</td>
<td>
<p style="margin-left: 1em;">EMIM: Estimation of Maternal, Imprinting and interaction effects using Multinomial modelling</p>
</td></tr>
</table>
</div>
<!-- End Header -->
<!-- Begin Menu Navigation -->
<ul id="menunav">
<li><a href="index.html">Home</a></li>
<li><a href="downloads.html">Download</a></li>
<li><a href="contact.html">Contact</a></li>
</ul>
<!-- End Menu Navigation -->

</td></tr>
<tr><td>
<table border="0" cellpadding="0" cellspacing="0"><tr>
<td valign="top">
<!-- Begin Left Column -->
<div id="leftcolumn">
<!-- Begin Contents Navigation -->
<div id="navcontainer">
<ul>
<li><a href="introduction.html">1 Introduction</a>
<ul>
<li><a href="introduction.html#information">1.1 Program information and citation</a></li>
</ul>
</li>
<li><a href="quick-start.html">2 Quick Start</a>
</li>
<li><a href="comp.html">3 Slow Start</a>
<ul>
<li><a href="comp.html#comp-premim">3.1 Compilation of PREMIM source code</a></li>
<li><a href="comp.html#comp-emim">3.2 Compilation of EMIM source code</a></li>
</ul>
</li>
<li><a href="premim.html">4 PREMIM: Pedigree file conversion</a>
<ul>
<li><a href="premim.html#premim-usage">4.1 Usage</a></li>
<li><a href="premim.html#premim-input">4.2 Input files</a></li>
<li><a href="premim.html#premim-output">4.3 Output files</a></li>
<li><a href="premim.html#premim-processing">4.4 Data processing</a></li>
<li><a href="premim.html#premim-parallel">4.5 Parallel processing</a></li>
<li><a href="premim.html#risk-allele">4.6 Changing the risk allele</a></li>
</ul>
</li>
<li><a href="emim.html">5 EMIM: Input files</a>
<ul>
<li><a href="emim.html#emimmarkers">5.1 emimmarkers.dat</a></li>
<li><a href="emim.html#emimparams">5.2 emimparams.dat</a></li>
<li><a href="emim.html#caseparenttrios">5.3 caseparenttrios.dat</a></li>
<li><a href="emim.html#caseparents">5.4 caseparents.dat</a></li>
<li><a href="emim.html#casemotherduos">5.5 casemotherduos.dat</a></li>
<li><a href="emim.html#casefatherduos">5.6 casefatherduos.dat</a></li>
<li><a href="emim.html#casemothers">5.7 casemothers.dat</a></li>
<li><a href="emim.html#casefathers">5.8 casefathers.dat</a></li>
<li><a href="emim.html#cases">5.9 cases.dat</a></li>
<li><a href="emim.html#conparents">5.10 conparents.dat</a></li>
<li><a href="emim.html#conmotherduos">5.11 conmotherduos.dat</a></li>
<li><a href="emim.html#confatherduos">5.12 confatherduos.dat</a></li>
<li><a href="emim.html#cons">5.13 cons.dat</a></li>
</ul>
</li>
<li><a href="output.html">6 EMIM: Usage and output files</a>
<ul>
<li><a href="output.html#alt-paras">6.1 Alternative parameterisations</a></li>
</ul>
</li>
<li><a href="example.html">7 Example using PREMIM and EMIM</a>
<ul>
<li><a href="example.html#eg-run-premim">7.1 Running PREMIM</a></li>
<li><a href="example.html#eg-run-emim">7.2 Running EMIM</a></li>
</ul>
</li>
<li><a href="parallel.html">8 Parallel processing</a>
</li>
<li><a href="poo.html">9 Parent-of-origin effects with SHAPEIT2</a>
<ul>
<li><a href="poo.html#options">9.1 Options</a></li>
<li><a href="poo.html#files">9.2 Data files</a></li>
<li><a href="poo.html#poo-example">9.3 Example Analysis</a></li>
</ul>
</li>
<li><a href="trouble.html">10 Troubleshooting and results interpretation</a>
<ul>
<li><a href="trouble.html#GWAS">10.1 GWAS data</a></li>
<li><a href="trouble.html#Imputed">10.2 Imputed data</a></li>
<li><a href="trouble.html#Merged">10.3 Merged data</a></li>
</ul>
</li>
<li><a href="references.html">References</a></li>
</ul>
</div>
<!-- End Contents Navigation -->
</div>
<!-- End Left Column -->
</td>
<td valign="top">
<!-- Begin Right Column -->
<div id="rightcolumn">
<h1>6 EMIM: Usage and output files</h1>
<p>
The program is run by typing &ldquo;./emim&rdquo;. This should produce 2 primary output files: <i>emimresults.out</i> and <i>emimsummary.out</i>. </p>
<p>
In previous versions of EMIM, two additional files (<i>fort.11</i> and <i>fort.12</i>) with details of the maximisation procedure were also produced by the MAXFUN subroutine. These files can be quite large and may subsequently be deleted if detailed output is not required. (However they can be quite useful for troubleshooting when the maximisation procedure does not appear to have worked correctly). In current versions of EMIM we do not output these files by default. Please contact us if you need a version of EMIM that outputs these files. </p>
<p>
The main results are given in the file <i>emimresults.out</i>. A fairly large number of lines of results are output for each SNP in turn, therefore this file can become quite large. If you particularly want to see the detailed results for some specific SNPs, we recommend you use PLINK to create a smaller subset containing just the SNPs of interest, and then re-run PREMIM and EMIM on this subset. </p>
<p>
The file <i>emimresults.out</i> first shows the parameter estimates from a global null model (in which the parameters <img src="http://latex.codecogs.com/png.latex?\inline R_1 \small " alt="R_1"/>, <img src="http://latex.codecogs.com/png.latex?\inline R_2 \small " alt="R_2"/>, <img src="http://latex.codecogs.com/png.latex?\inline S_1 \small " alt="S_1"/>, <img src="http://latex.codecogs.com/png.latex?\inline S_2 \small " alt="S_2"/>, <img src="http://latex.codecogs.com/png.latex?\inline I_m \small " alt="I_m"/>, <img src="http://latex.codecogs.com/png.latex?\inline I_p \small " alt="I_p"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{11} \small " alt="\gamma_{11}"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{12} \small " alt="\gamma_{12}"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{21} \small " alt="\gamma_{21}"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{22} \small " alt="\gamma_{22}"/> are all set to equal 1.0). Next the parameter estimates from the specified alternative model (as specified in the file <i>emimparams.dat</i>) are given, both on their original and log scales, followed by 95% confidence intervals for estimated parameters of interest. Finally the maximized ln likelihoods for the alternative and null models are given, together with twice the difference between these (which can be compared to a chi-squared on the appropriate df (=the number of estimated parameters of interest) to calculate a p-value, if required). Note that if nested alternative (non-null) models are to be compared, you will have to run EMIM twice and compare twice the difference between the maximized ln likelihoods for each of the alternative models to a chi-squared on the appropriate df. </p>
<p>
A summary of the results from <i>emimresults.out</i> is given in the file <i>emimsummary.out</i>. This file can be more convenient to deal with (e.g. by reading in and manipulating in R) if analysis is being performed on a large number of SNPs. First comes a header line describing the different columns. Then comes a single line of results for each SNP analysed. The entries in each line correspond to the SNP number, the SNP ID, the estimated allele frequency (or input allele frequency if allele frequency is not estimated), then for each parameter (<img src="http://latex.codecogs.com/png.latex?\inline R_1 \small " alt="R_1"/>, <img src="http://latex.codecogs.com/png.latex?\inline R_2 \small " alt="R_2"/>, <img src="http://latex.codecogs.com/png.latex?\inline S_1 \small " alt="S_1"/>, <img src="http://latex.codecogs.com/png.latex?\inline S_2 \small " alt="S_2"/>, <img src="http://latex.codecogs.com/png.latex?\inline I_m \small " alt="I_m"/>, <img src="http://latex.codecogs.com/png.latex?\inline I_p \small " alt="I_p"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{11} \small " alt="\gamma_{11}"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{12} \small " alt="\gamma_{12}"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{21} \small " alt="\gamma_{21}"/>, <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{22} \small " alt="\gamma_{22}"/>) in turn, we have the estimate of the logarithm of the relevant parameter, its estimated standard error, and the estimated lower and upper 95% confidence limit for the logarithm of the relevant parameter. Finally we have the maximized log likelihood under the null, the maximized log likelihood under the alternative, and twice the difference between these, followed by a column indicating whether there were any warning messages for this SNP output to the file <i>emimresults.out</i> (in which case we recommend you take a look at the file <i>emimresults.out</i> to see what the warning messages say). </p>
<p>
<b>PLEASE NOTE THAT EMIM DOES NOT ITSELF GENERATE ANY P-VALUES. Instead, EMIM produces chi-squared statistics and maximized log likelihoods (that can be compared for nested models and used to create p-values, see Example section below). In order to calculate the p-value for any test, you will therefore need to compare the chi-squared statistic or twice the difference in maximized log likelihoods to a chi-squared on the appropriate degrees of freedom, using an external program such as Excel or R.</b> </p>
<h2><a name="alt-paras">6.1 Alternative parameterisations</a></h2>
<p>
If one of the parameterizations from the third section of <i>emimparams.dat</i> has been selected, the output parameters in <i>emimsummary.out</i> are slightly different. If the <a href="references.html#sinsheimer:etal:03">Sinsheimer et al. (2003)</a> MFG interaction parameters have been chosen, then instead of <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{11} \small " alt="\gamma_{11}"/> and/or <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{22} \small " alt="\gamma_{22}"/>, you should find that MFG parameters <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{01} \small " alt="\gamma_{01}"/> and/or <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{21} \small " alt="\gamma_{21}"/> (denoted MFG01 and MFG21) are output. If the <a href="references.html#palmer:etal:06">Palmer et al. (2006)</a> interaction parameter has been chosen, then instead of <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{11} \small " alt="\gamma_{11}"/> you should find the Palmer match parameter (denoted MFGmu) is output. If the <a href="references.html#li:etal:09">Li et al. (2009)</a> and <a href="references.html#parimi:etal:08">Parimi et al. (2008)</a> interaction parameter has been chosen, then instead of <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{22} \small " alt="\gamma_{22}"/>, you should find that the <a href="references.html#li:etal:09">Li et al. (2009)</a> conflict parameter (denoted Jc) is output. Hopefully the header line in <i>emimsummary.out</i> should make it clear which parameters have been output. </p>
<p>
Moreover, if one of the parameterizations from the third section of <i>emimparams.dat</i> has been selected, the value of the other estimated parameters may be different. For example, if the <a href="references.html#sinsheimer:etal:03">Sinsheimer et al. (2003)</a> MFG interaction parameterisation is chosen, the R1 and R2 parameters as output by EMIM will actually correspond to what were called <img src="http://latex.codecogs.com/png.latex?\inline \rho_1 \small " alt="\rho_1"/> and <img src="http://latex.codecogs.com/png.latex?\inline \rho_2 \small " alt="\rho_2"/> in Table I (Example 9) of <a href="references.html#ainsworth:etal:11">Ainsworth et al. (2011)</a>, and <img src="http://latex.codecogs.com/png.latex?\inline S_1 \small " alt="S_1"/> and <img src="http://latex.codecogs.com/png.latex?\inline S_2 \small " alt="S_2"/> will actually correspond to what were called <img src="http://latex.codecogs.com/png.latex?\inline \eta_1 \small " alt="\eta_1"/> and <img src="http://latex.codecogs.com/png.latex?\inline \eta_2 \small " alt="\eta_2"/> in Table I (Example 9) of <a href="references.html#ainsworth:etal:11">Ainsworth et al. (2011)</a>. Perusal of Table I (Example 9) of <a href="references.html#ainsworth:etal:11">Ainsworth et al. (2011)</a> shows that <img src="http://latex.codecogs.com/png.latex?\inline \rho_1 \small " alt="\rho_1"/> (<img src="http://latex.codecogs.com/png.latex?\inline \rho_2 \small " alt="\rho_2"/>) is the factor by which an individual's disease risk is multiplied if they possess one (two) risk alleles at a given locus, and <img src="http://latex.codecogs.com/png.latex?\inline \eta_1 \small " alt="\eta_1"/> (<img src="http://latex.codecogs.com/png.latex?\inline \eta_2 \small " alt="\eta_2"/>) is the factor by which an individual's disease risk is multiplied if their mother possesses one risk allele at that locus, when using the Sinsheimer parameterisation. That is why these parameters are still denoted <img src="http://latex.codecogs.com/png.latex?\inline R_1 \small " alt="R_1"/>, <img src="http://latex.codecogs.com/png.latex?\inline R_2 \small " alt="R_2"/>, <img src="http://latex.codecogs.com/png.latex?\inline S_1 \small " alt="S_1"/> and <img src="http://latex.codecogs.com/png.latex?\inline S_2 \small " alt="S_2"/> in the EMIM output - because they have the same meaning that we are attributing to <img src="http://latex.codecogs.com/png.latex?\inline R_1 \small " alt="R_1"/>, <img src="http://latex.codecogs.com/png.latex?\inline R_2 \small " alt="R_2"/>, <img src="http://latex.codecogs.com/png.latex?\inline S_1 \small " alt="S_1"/> and <img src="http://latex.codecogs.com/png.latex?\inline S_2 \small " alt="S_2"/> when using the default interaction parameterisation (i.e. when interaction is modelled in terms of parameters <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{11} \small " alt="\gamma_{11}"/> and <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{22} \small " alt="\gamma_{22}"/>, instead of <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{01} \small " alt="\gamma_{01}"/> and <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{21} \small " alt="\gamma_{21}"/>). However, the values of <img src="http://latex.codecogs.com/png.latex?\inline R_1 \small " alt="R_1"/>, <img src="http://latex.codecogs.com/png.latex?\inline R_2 \small " alt="R_2"/>, <img src="http://latex.codecogs.com/png.latex?\inline S_1 \small " alt="S_1"/> and <img src="http://latex.codecogs.com/png.latex?\inline S_2 \small " alt="S_2"/> will not necessarily be the same within these two paramaterisations. See <a href="references.html#ainsworth:etal:11">Ainsworth et al. (2011)</a> more discussion of this issue. </p>
<p>
So, in summary, if using the <a href="references.html#sinsheimer:etal:03">Sinsheimer et al. (2003)</a> MFG interaction parameterisation, the parameters <img src="http://latex.codecogs.com/png.latex?\inline R_1 \small " alt="R_1"/> / <img src="http://latex.codecogs.com/png.latex?\inline R_2 \small " alt="R_2"/> / <img src="http://latex.codecogs.com/png.latex?\inline S_1 \small " alt="S_1"/> / <img src="http://latex.codecogs.com/png.latex?\inline S_2 \small " alt="S_2"/> / <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{01} \small " alt="\gamma_{01}"/> / <img src="http://latex.codecogs.com/png.latex?\inline \gamma_{21} \small " alt="\gamma_{21}"/> estimated by EMIM are actually equivalent to the parameters that were called <img src="http://latex.codecogs.com/png.latex?\inline \rho_1 \small " alt="\rho_1"/> / <img src="http://latex.codecogs.com/png.latex?\inline \rho_2 \small " alt="\rho_2"/> / <img src="http://latex.codecogs.com/png.latex?\inline \eta_1 \small " alt="\eta_1"/> / <img src="http://latex.codecogs.com/png.latex?\inline \eta_2 \small " alt="\eta_2"/> / <img src="http://latex.codecogs.com/png.latex?\inline \mu_0 \small " alt="\mu_0"/> / <img src="http://latex.codecogs.com/png.latex?\inline \mu_2 \small " alt="\mu_2"/> by <a href="references.html#sinsheimer:etal:03">Sinsheimer et al. (2003)</a>. </p>
<br />
<div class="prevnext"><span class="left"><a href="emim.html">&lt;-prev</a>
</span><span class="right"><a href="example.html">next-&gt;</a>
</span></div><!-- End Right Column -->
</div>
</td>
</tr></table>
<!-- End Wrapper -->
</td></tr></table>
</div>
 <!-- Start of StatCounter Code --> <script type="text/javascript"> var sc_project=6887608; var sc_invisible=1; var sc_security="98b151c3"; </script> <script type="text/javascript" src="http://www.statcounter.com/counter/counter.js"></script><noscript><div class="statcounter"><a title="weebly reliable statistics" href="http://statcounter.com/weebly/" target="_blank"><img class="statcounter" src="http://c.statcounter.com/6887608/0/98b151c3/1/" alt="weebly reliable statistics" ></a></div></noscript> <!-- End of StatCounter Code --></body>
</html>
